.github/workflows/clean_caches.yml

```diff
@@ -34,18 +31,19 @@ jobs:
     if: github.event_name == 'workflow_dispatch' || github.event_name == 'schedule'
     permissions: write-all
     runs-on: ubuntu-latest
-    steps:        
+    steps: 
       - run: |
           gh extension install actions/gh-actions-cache
-          tarballs=$(gh actions-cache list -R ${{ github.repository }} --key tar- | cut -f 1 )
 
           # Even when we name the cache location explicitly, 
           # caches for the layers will be stored under buildkit-blobs.
-          # We sort them by size, with the largest ones first.          
-          buildkits=$(gh actions-cache list -R ${{ github.repository }} --key buildkit-blob --sort size --order desc | cut -f 1 )
+          # We sort them by size, with the largest ones first.        
+          buildkits=$(gh actions-cache list -R ${{ github.repository }} --key buildkit-blob --sort size --order desc | cut -f 1)
+          tarballs=$(gh actions-cache list -R ${{ github.repository }} --key tar- | cut -f 1)
 
-          # For now, we keep the 9 largest buildkit caches.
-          for key in "${buildkits[@:9]} ${tarballs[@]}"
+          # For now, we keep the 9 largest caches.
```

Comment:
I'm not sure why the size of the cached blob is used here. Shouldn't we be saving something like the 9 latest tarballs?

Reply:
Right now, there are no tarballs pushed into the cache.
The reason to push a tarball to the cache is because as silly as it might be more convenient for sharing between jobs than other options. In that case, however, that cache ideally should be deleted once all jobs finish. Unfortunately, CI doesn't have the necessary permissions to do that for PRs from forks. 
The bottom line is
1) cache clean up may require some further tweaks (e.g. with docs generation being added)
2) build caches and images pushed to registries are the two things intended to be reused beyond a singe workflow.

Reply:
As to why size opposed to latest: Due to how the GitHub cache works, the buildkit produces very small caches pretty much on every build. These rely on the much bigger cache that is the actual build of the image layer that may have been produced a lot longer back. I would hence need to either figure out a way to walk dependencies, or I make sure the large caches are the ones that stick around.